.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env bash
+set -e
+
+# GitHub repository details
+ORG="redhat-appstudio"
+REPO="release-service-catalog"
+
+OPTIONS=$(getopt --long "target-branch:,fork-owner:,force-to-staging,override,debug,help" -o "tgt:,fo:,h" -- "$@")
+eval set -- "$OPTIONS"
+while true; do
+  case "$1" in
+    -tgt|--target-branch)
+        TARGET_BRANCH="$2"
+        shift 2
+        ;;
+    -fo|--fork-owner)
+        FORK_OWNER="$2"
+        shift 2
+        ;;
+    --force-to-staging)
+        FORCE_TO_STAGING="true"
+        shift
+        ;;
+    --override)
+        OVERRIDE="true"
+        shift
+        ;;
+    --debug)
+        DEBUG="true"
+        shift
+        ;;
+    -h|--help)
+        print_help
+        exit
+        ;;
+    --)
+        shift
+        break
+        ;;
+    *) echo "Error: Unexpected option: $1" % >2
+  esac
+done
+
+print_help(){
+  echo -e "$0 --target-branch TARGET_BRANCH --fork-owner FORK_OWNER\n"
+  echo -e "\t--target-branch TARGET_BRANCH\tName of the overlay to target for promotion"
+  echo -e "\t--fork-owner FORK_OWNER\tName of the owner of your release-service-catalog fork in Github"
+  echo -e "\t--force-to-staging\tIf passed, allow promotion to staging even if staging and production differ"
+  echo -e "\t--override\tIf passed, allow promotion to production even if the change has not been in staging for one week"
+  echo -e "\t--debug\tFor debugging purposes, do everything except for creating the PR and deleting the temp repo"
+}
+
+check_if_branch_differs() {
```

Comment:
Here's a suggestion. What if we applied the branch change in the yaml files before checking? That way you could just check that they're really the same:
```sh
SOURCE=$1
TARGET=$2

for PIPELINE in $(ls pipelines)
do
    PIPELINE_FILE=pipelines/$PIPELINE/${PIPELINE}.yaml
    yq . $PIPELINE_FILE > ${PIPELINE_FILE}.1
    yq '.spec.params[] |= select(.name == "taskGitRevision") |= .default = "'$TARGET'"'\
        $PIPELINE_FILE > ${PIPELINE_FILE}.2
    diff ${PIPELINE_FILE}.1 ${PIPELINE_FILE}.2 > ${PIPELINE_FILE}.diff
    patch $PIPELINE_FILE < ${PIPELINE_FILE}.diff
done

```
(Adapted from https://stackoverflow.com/a/57938189 )

Then `git diff origin/$1` should really be empty except for the README changes - it would be good to get rid of those too - maybe just agree to remove that default value there? Or include a generic extra note below the table to say that the default is based on the branch you are in? That way we wouldn't have to deal with that one.

Reply:
https://issues.redhat.com/browse/RHTAPREL-788

Reply:
READMEs are adjusted and I added the check you suggested instead of using xargs and sed

Reply:
Cool. Looks good. You probably guessed it, but the reason for this cumbersome diff manipulation is that `yq` can (and will) change some formatting in the file as a side effect, so this overcomes this. Maybe add a comment about it somewhere there? 

(Also, this could have been useful for Leandro's stuff a while back - he was dealing with exactly the same issue - changing formatting of a yaml as a side effect of changing a field's value - I didn't realize then you could overcome it quite easily like this.)

Reply:
comment added

Reply:
Awesome, thanks.